docker_setup.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python2
+from __future__ import print_function
+
+import os
+import os.path
+
+from os import symlink
+from os.path import exists, join
+from shlex import split
+from sys import exit
+from subprocess import call
+
+def link_or_generate_ssh_keys():
+    if 'SSH_KEY' not in os.environ:
+        os.environ['SSH_KEY'] = 'id_rsa'
+
+    ssh_key = join(os.environ['MANTL_CONFIG_DIR'], os.environ['SSH_KEY'])
+    if not exists(ssh_key):
+        call(split('ssh-keygen -N "" -f {}'.format(ssh_key)))
+
+    print("Symlinking {} to /root/.ssh/id_rsa".format(ssh_key))
+    symlink(ssh_key, '/root/.ssh/id_rsa')
+
+    ssh_key += '.pub'
+    print("Symlinking {} to /root/.ssh/id_rsa.pub".format(ssh_key))
+    symlink(ssh_key, '/root/.ssh/id_rsa.pub')
+
+
+def link_terraform_files():
+    cfg_d = os.environ['MANTL_CONFIG_DIR']
+    tfs = [join(cfg_d, f) for f in os.listdir(cfg_d) if f.endswith('.tf')]
+    if len(tfs):
+        for tf in tfs:
+            base = os.path.basename(tf)
+            print("Symlinking {} to {}".format(tf, base))
+            symlink(tf, base)
+    else:
+        if 'MANTL_PROVIDER' not in os.environ:
+            print("mantl.readthedocs.org for provider")
+            exit(1)
+        tf = 'terraform/{}.sample.tf'.format(os.environ['MANTL_PROVIDER'])
+
+        print("Symlinking {} to mantl.tf".format(tf))
+        symlink(tf, 'mantl.tf')
+
+
+def link_ansible_playbook():
+    ansible_playbook = join(os.environ['MANTL_CONFIG_DIR'], 'mantl.yml')
+    if not exists(ansible_playbook):
+        ansible_playbook = 'sample.yml'
+
+    print("Symlinking {} to mantl.yml".format(ansible_playbook))
+    symlink(ansible_playbook, 'mantl.yml')
+
+
+def link_or_generate_security_file():
+    security_file = join(os.environ['MANTL_CONFIG_DIR'], 'security.yml')
```

Comment:
There are two possible mount points from the host into this container: `/mantl`, and `/local` (or whatever MANTL_CONFIG_DIR is set to). All of our code makes the assumption that the root of the repository is where you run commands. However, I want to make sure that this workflow persists state between container runs; therefore, the user mounts the `/local` volume so that `tfstate` and other needed files persist. This also allows users to load in custom configuration.